test/integration/smoke/test_activate_template.py

```diff
@@ -68,17 +64,31 @@ def setUpClass(cls):
 
         cls.listTemplatesCmd = listTemplates.listTemplatesCmd()
         cls.listTemplatesCmd.templatefilter= 'system'
-        # Save the system templates
+        # Save the current system template
         cls.templates = cls.apiclient.listTemplates(cls.listTemplatesCmd)
+        cls.firstRun = True
     
     @classmethod
     def tearDownClass(cls):
         try:
-            # Clean up the created templates
-            for temp in cls.templates:
-                cmd = activateSystemVMTemplate.activateSystemVMTemplateCmd()
-                cmd.id = temp.id
-                cls.apiclient.activateSystemVMTemplate(cmd)
+            if cls.firstRun:
+                # Save currently active template
+                activeTemplate = cls.apiclient.listTemplates(cls.listTemplatesCmd)[0]
+                cmd = deleteTemplate.deleteTemplateCmd()
```

Comment:
deleteTemaplate?

Reply:
Yes, we need to delete the newly created template as part of the cleanup.